rules/msk_topic_config_comments.go

```diff
@@ -0,0 +1,317 @@
+	if err != nil {
+		return err
+	}
+	if msg == "" {
+		return nil
+	}
+
+	comment, err := r.getExistingComment(runner, timePair)
+	if err != nil {
+		return err
+	}
+
+	if comment == nil {
+		err := runner.EmitIssueWithFix(
+			r,
+			fmt.Sprintf("%s must have a comment with the human readable value: adding it ...", configValueInfo.key),
+			timePair.Key.Range(),
+			func(f tflint.Fixer) error {
+				return f.InsertTextAfter(timePair.Value.Range(), msg)
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("emitting issue: no comment for time value: %w", err)
+		}
+		return nil
+	}
+
+	commentTxt := strings.TrimSpace(string(comment.Bytes))
+	if commentTxt != msg {
+		issueMsg := fmt.Sprintf(
+			"%s value doesn't correspond to the human readable value in the comment: fixing it ...",
+			configValueInfo.key,
+		)
+		err := runner.EmitIssueWithFix(r, issueMsg, comment.Range,
+			func(f tflint.Fixer) error {
+				return f.ReplaceText(comment.Range, msg+"\n")
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("emitting issue: wrong comment for time value: %w", err)
+		}
+	}
+	return nil
+}
+
+func (r *MSKTopicConfigCommentsRule) getExistingComment(
+	runner tflint.Runner,
+	pair hcl.KeyValuePair,
+) (*hclsyntax.Token, error) {
+	comments, err := r.getCommentsForFile(runner, pair.Key.Range().Filename)
+	if err != nil {
+		return nil, err
+	}
+
+	// first look for the comment on the same line, after the property definition.
+	// Example: "retention.ms" = "2629800000" # keep data for 30 days
+	afterPropertyIdx := slices.IndexFunc(comments, func(comment hclsyntax.Token) bool {
+		return comment.Range.Start.Line == pair.Key.Range().Start.Line &&
+			comment.Range.Start.Column > pair.Value.Range().End.Column
+	})
+
+	if afterPropertyIdx >= 0 {
+		return &comments[afterPropertyIdx], nil
+	}
+
+	/* second, look for the comment on the previous line, before the property definition. Example:
+	# keep data for 30 days
+	"retention.ms" = "2629800000"
+	*/
+	beforePropertyIdx := slices.IndexFunc(comments, func(comment hclsyntax.Token) bool {
+		return comment.Range.Start.Line == pair.Key.Range().Start.Line-1 &&
+			comment.Range.End.Line == pair.Key.Range().Start.Line
+	})
+	if beforePropertyIdx >= 0 {
+		return &comments[beforePropertyIdx], nil
+	}
+
+	return nil, nil
+}
+
+func (r *MSKTopicConfigCommentsRule) getCommentsForFile(
+	runner tflint.Runner,
+	filename string,
+) (hclsyntax.Tokens, error) {
+	// we need to parse the file every time, otherwise keeping a cache per file doesn't work
+	file, err := runner.GetFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("getting hcl file %s for reading comments: %w", filename, err)
+	}
+
+	tokens, diags := hclsyntax.LexConfig(file.Bytes, filename, hcl.InitialPos)
+	if diags != nil {
+		return nil, diags
+	}
+
+	return slices.DeleteFunc(tokens, isNotComment), nil
+}
+
+func isNotComment(token hclsyntax.Token) bool {
+	return token.Type != hclsyntax.TokenComment
+}
```

Comment:
Nit/subjective: could reduce scope of function

```suggestion
	isNotComment := func(token hclsyntax.Token) bool {
		return token.Type != hclsyntax.TokenComment
	}

	return slices.DeleteFunc(tokens, isNotComment), nil
}
```

Reply:
:) I had it like that before and refactored to be more static... just like extracting a constant. I would prefer to keep it like this.